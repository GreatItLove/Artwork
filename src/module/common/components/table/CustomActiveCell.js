import React from 'react';
import FontIcon from 'material-ui/FontIcon';
import PropTypes from 'prop-types';

const propTypes = {
  value: PropTypes.bool.isRequired
};

/**
 * @function CustomActiveCell
 * @param {String} value
 * @returns {XML}
 */
function CustomActiveCell({ value }) {
  const content = value ? (
    <FontIcon className="material-icons">done</FontIcon>
  ) : (
    <FontIcon className="material-icons">clear</FontIcon>
  );
  return content;
}

CustomActiveCell.propTypes = propTypes;

export default CustomActiveCell;
