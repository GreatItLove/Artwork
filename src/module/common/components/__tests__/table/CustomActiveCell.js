import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import CustomActiveCell from '../../table/CustomActiveCell';

function createWrapper(value) {
  return shallow(<CustomActiveCell value={value} />);
}

describe('CustomActiveCell', () => {
  it('Snapshot with true value', () => {
    const wrapper = createWrapper(true);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('Snapshot with false value', () => {
    const wrapper = createWrapper(false);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
