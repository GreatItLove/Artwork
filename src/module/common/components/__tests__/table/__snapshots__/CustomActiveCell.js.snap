// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomActiveCell Snapshot with false value 1`] = `
<FontIcon
  className="material-icons"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  clear
</FontIcon>
`;

exports[`CustomActiveCell Snapshot with true value 1`] = `
<FontIcon
  className="material-icons"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  done
</FontIcon>
`;
